### PR TITLE
reference holonix

### PR DESCRIPTION
**Pull request by thedavidmeister**
## testing/benchmarking notes

( if any manual testing or benchmarking was/should be done, add notes and/or screenshots here )

## followups

( any new tickets/concerns that were discovered or created during this work but aren't in scope for review here )

## changelog

Please check one of the following, relating to the [CHANGELOG-UNRELEASED.md](https://github.com/holochain/holochain-rust/blob/develop/CHANGELOG-UNRELEASED.md)

- [ ] this is a code change that effects some consumer (e.g. zome developers) of holochain core so it is added to the CHANGELOG-UNRELEASED.md (linked above), with the format `- summary of change [PR#1234](https://github.com/holochain/holochain-rust/pull/1234)`
- [ ] this is not a code change, or doesn't effect anyone outside holochain core development

### CODE REVIEW

File: default.nix [hunk 1]
```diff
@@ -1,18 +1,19 @@
 let
+ holonix-release-tag = "expecting-bad-ozone";
+ holonix-release-sha256 = "0x6bma8p1kmlzsr2dd5zjyvw809hjmycdk8n8bknigx7vd88v52m";
 
-  pkgs = import ./holonix/nixpkgs/nixpkgs.nix;
-  darwin = import ./holonix/darwin/config.nix;
-  openssl = import ./holonix/openssl/config.nix;
-  rust = import ./holonix/rust/config.nix;
-
+ holonix = import (fetchTarball {
```
Comment (thedavidmeister) [inline default.nix]: this is where the magic happens

File: default.nix [hunk 2]
```diff
@@ -1,18 +1,19 @@
 let
+ holonix-release-tag = "expecting-bad-ozone";
```
Comment (thedavidmeister) [inline default.nix]: not using semver

**Comment by Connoropolous** [comment]
@thedavidmeister can we get a little explainer of what the outcomes of this PR are?

**Comment by thedavidmeister** [comment]
@Connoropolous 

- moved the holonix directory to the holonix repo
- reworked the holonix output slightly to be a set rather than a derivation
  - means that downstream consumers can provide a context relevant derivation and use holonix as a variable
  - means that downstream consumers can reference the holonix repo by URL as a tarball
  - core is the first downstream so `foo` becomes `holonix.foo`
  - node js will be the second downstream so we can pull it out
- plan for future is that holonix repo will be refactored to be only the "commons" and downstreams will implement their own workflows e.g. `hc-*`

**Comment by thedavidmeister** [comment]
@Connoropolous for example... the conflicts in dist right now should sit in core!

**Comment by Connoropolous** [comment]
thanks @thedavidmeister

**Pull request merged**